app/WebParsing/PostParser.hs

```diff
@@ -30,36 +32,64 @@ addPostToDatabase programElements = do
 
     case P.parse postInfoParser "POSt information" fullPostName of
         Left _ -> return ()
-        Right post -> do
-            postExists <- insertUnique post
+        Right (Post postType deptName code _) -> do
+            postExists <- insertUnique $ Post postType deptName code (T.pack $ intercalate "\n" $ map T.unpack $ concat requirementLines)
             case postExists of
                 Just key ->
                     mapM_ (insert_ . PostCategory key) requirements
                 Nothing -> return ()
     where
         isRequirementSection tag = tagOpenAttrNameLit "div" "class" (T.isInfixOf "views-field-field-enrolment-requirements") tag || tagOpenAttrNameLit "div" "class" (T.isInfixOf "views-field-field-completion-requirements") tag
 
-
 -- | Parse a Post value from its title.
 -- Titles are usually of the form "Actuarial Science Major (Science Program)".
 postInfoParser :: Parser Post
 postInfoParser = do
-    deptName <- parseDepartmentName
```

Comment:
Okay overall this parsing is getting too complicated. This was appropriate for the prerequisite parsing because that's all over the place in the calendar, but we really don't need this here.

Let's switch strategies and have the post name be the full name up to the final `" - "`, and the post code everything after that (which is the current behaviour). Then, parse the post type from the post code: the text from chars 2-4 map to the post type, `SPE` -> `Specialist`, etc.

Reply:
I implemented this, but there are five cases where the post string doesn't have a post code[^1]. In those cases, I made the parser fall back to look for the post type in the department name. Hope it is not too complex!

[^1]: "Focus in Green Chemistry"<br>"Focus in Practical Chemical Analysis"<br>"Focus in Data Science in Business"<br>"Focus in Managing in Diverse Economies (UofT Global Scholar)"<br>"Biological Physics Specialist"